chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/CertifiedProductManager.java

```diff
@@ -1202,6 +1209,17 @@ private void updateListingsChildData(CertifiedProductSearchDetails existingListi
         updateCqms(updatedListing, existingListing.getCqmResults(), updatedListing.getCqmResults());
     }
 
+    private void updateRwtEligibilityForListingAndChildren(Long listingId) {
+        rwtCachingService.calculateRwtEligibility(listingId);
+
+        List<CertifiedProductDTO> listingChildren = listingGraphDao.getChildren(listingId);
+        if (!CollectionUtils.isEmpty(listingChildren)) {
+            for (CertifiedProductDTO child : listingChildren) {
+                updateRwtEligibilityForListingAndChildren(child.getId());
+            }
+        }
+    }
+
```

Comment:
Could we maybe move this method or logic to the `RealWorldTestingEligiblityCachingService`.  That way if we need to perform this task outside this Manager, it would be available.

Reply:
I am not sure how to move this and have it work... the method "calculateRwtEligibility" in the RealWorldTestingEligiblityCachingService has a CachePut annotation. This method is called recursively from itself so if it were all in the RealWorldTestingEligiblityCachingService class I think it would end up ignoring the CachePut annotation at because of Spring proxies. Am I thinking about it wrong?